Search Bar Implementation & Login Routing Fix

hopieyimmie opened the pull request:
## Developer: Hope Yim

Closes #18 

### Pull Request Summary

From Aidan's implementation of the Client home page, I created a filtering way so the user is able to search any client by their name, address, or number. I was also able to display the name of the client by changing headofHousehold in the props to first and last name. In addition, when logging into the site, I changed the routing to go to /homePage instead of the /clientPage.

### Modifications

- page.tsx from /login
- page.tsx from /homePage
- clientInformationTable.tsx from /components
- package.json but when reviewing the PR it's just the Mongo one (I had to do a npm mongo install because of my env file but I think it can be ignored)

### Testing Considerations

Please verify by searching up a user in the database by name, number, and address.
There are 2 things to keep in mind when testing:
- You can search up a user on the first page but not on the second page (will create debugging task)
- The error statement still appears (will create debugging task as well)

I will create debugging tasks once this is approved and merged.

For the routing one, just check by logging in and see if it leads you to the home page instead of the client page.

### Pull Request Checklist

- [x] Code is neat, readable, and works
- [x] Comments are appropriate
- [x] The commit messages follows our [guidelines](https://h4i.notion.site/Conventional-Commits-593452ad1179489399ad3bd696ef772a)
- [x] The developer name is specified
- [x] The summary is completed
- [x] Assign reviewers

### Screenshots/Screencast
<img width="1440" alt="Screen Shot 2024-05-06 at 9 41 42 PM" src="https://github.com/hack4impact-calpoly/five-cities-food-pantry/assets/47905779/7c961f94-049a-41e7-8cf2-9b8385c85004">

ajesus7 commented:
### Slight notes for possible improvements
- One note for change: when you type in a search query, it automatically applies the search (after each typed character) and the results are filtered accordingly. However, when you type in a search and then click the search button (magnifying glass), the page seems to be refreshed and the search query is lost, resulting in all the users being displayed again. We should possibly get rid of the magnifying glass so that the user is not tempted to treat it like a button (although the magnifying glass does nicely signify the search feature), change the functionality so that the clients are only filtered when the button is clicked, or find some other solution to this slight UX problem :) 
- In response to your testing consideration about error message still appearing, I agree! If a search results in no users, the error message displayed should be more semantically related to the error; instead of displaying the current error message, "Error: There are no clients associated with this user.", which occurs when there are no clients, we should adjust it so that it says something like "Error: There are no clients that match your search".

### Below are other comments about things that worked well!

- Also, I was able to search for users on the second page! (Melika and Michael Walker).
- I was able to search for users by each parameter (address, number, name).
- The routing from login to /homePage worked!

Good job!!

hopieyimmie commented:
Improvements and Changes:
- Moved the error message (no clients one) to the bottom so it doesn't appear when refreshed
- Added error message when no client is found and matches the search
- Added preventDefault() for the search button so it prohibits the page to refresh when clicked

For some reason, when I try searching melika or michael on the second page it doesn't work for me. But if it works for you then that's great! Maybe it's just me.

Let me know if anything else needs to be changed! Thanks!
